apps/sms/views/inbox/js/inbox.js

```diff
@@ -697,77 +699,95 @@ var InboxView = {
-      // If the draft is newer than the message, update
-      // the body with the draft content's first string.
-      if (draft.timestamp >= record.timestamp) {
-        bodyHTML = draft.content.find(function(content) {
-          if (typeof content === 'string') {
-            return true;
-          }
-        });
-        type = draft.type;
-      }
-    }
+    var hasDraft = false;
+    var body = record.body;
+    var type = record.lastMessageType;
 
-    bodyHTML = Template.escape(bodyHTML || '');
+    var iconLabel = '';
 
-    li.id = 'thread-' + id;
-    li.dataset.threadId = id;
-    li.dataset.time = timestamp;
-    li.dataset.lastMessageType = type;
-    li.classList.add('threadlist-item');
+    // If the draft is newer than the message, update
+    // the body with the draft content's first string.
+    if (draft && draft.timestamp >= record.timestamp) {
+      body = draft.content.find(function(content) {
+        if (typeof content === 'string') {
+          return true;
+        }
+      });
+      type = draft.type;
+      hasDraft = true;
+    }
+
+    node.dataset.lastMessageType = type;
 
     if (draft) {
       // Set the "draft" visual indication
-      li.classList.add('draft');
+      node.classList.add('draft');
 
       if (isDraft) {
-        li.dataset.draftId = draft.id;
-        li.classList.add('is-draft');
+        node.dataset.draftId = draft.id;
+        node.classList.add('is-draft');
         iconLabel = 'is-draft';
       } else {
-        li.classList.add('has-draft');
+        node.classList.add('has-draft');
         iconLabel = 'has-draft';
       }
+    } else {
+      // remove it
+      node.classList.remove('draft');
     }
 
     if (record.unreadCount > 0) {
-      li.classList.add('unread');
+      node.classList.add('unread');
       iconLabel = 'unread-thread';
+    } else {
+      node.classList.remove('unread');
+    }
+
+    node.querySelector('.js-conversation-body').textContent = body || '';
+
+    var stateIndicator = node.querySelector('.js-conversation-state');
+    if (iconLabel) {
+      navigator.mozL10n.setAttributes(stateIndicator, iconLabel);
+    } else {
+      stateIndicator.removeAttribute('data-l10n-id');
     }
+  },
+
+  createThread: function inbox_createThread(record) {
+    // Create DOM element
+    var li = document.createElement('li');
+    var timestamp = +record.timestamp;
+    var id = record.id;
+    var thread = Threads.get(id);
+    // A "new message" draft
+    var isDraft = thread.isDraft;
+    // An existing conversation's draft.
+    var draft = thread.getDraft();
+
+    li.id = 'thread-' + id;
+    li.dataset.threadId = id;
+    li.dataset.time = timestamp;
+    li.classList.add('threadlist-item');
 
     // Render markup with thread data
     li.innerHTML = this.tmpl.thread.interpolate({
       hash: isDraft ? '#/composer' : '#/thread?id=' + id,
       mode: isDraft ? 'drafts' : 'threads',
       id: isDraft ? draft.id : id,
-      number: number,
```

Comment:
mmm it was not accidental, but I don't remember (the joy of developing a patch over long periods of time :p). I'll look at this, thanks !